src/v/v8_engine/environment.cc

```diff
@@ -14,17 +14,28 @@
 
 namespace v8_engine {
 
-enviroment::enviroment() {
+v8_platform_wrapper::v8_platform_wrapper() {
     v8::V8::SetFlagsFromString("--single-threaded");
     _platform = v8::platform::NewSingleThreadedDefaultPlatform();
     vassert(_platform.get() != nullptr, "Can not init platform for v8");
     v8::V8::InitializePlatform(_platform.get());
     v8::V8::Initialize();
 }
 
-enviroment::~enviroment() {
+v8_platform_wrapper::~v8_platform_wrapper() {
     v8::V8::Dispose();
     v8::V8::ShutdownPlatform();
 }
 
+enviroment::enviroment(size_t threads_num, uint8_t cpu_id)
+  : _executor(threads_num, cpu_id) {}
+
+enviroment::~enviroment() {
+    vassert(_executor.is_stopping(), "Executor was not stopped");
```

Comment:
Maybe check `is_stopped()` meaning that there are 0 jobs active. `is_stopping` means its in the process of shutting down and there still may be active jobs

Reply:
But it ok
I wiil wait all threads in executor destructor . I only need to know does anybody run stop for executor

Reply:
Delete it